File: frontend/src/api/notifications.ts
import { api } from './client'

export interface AppNotification {
  id: number
  type: string
  title: string
  body?: string | null
  meta?: string | null
  created_at: string
  read_at?: string | null
}

export const notificationsApi = {
  list:         () => api.get<AppNotification[]>('/notifications'),
  unreadCount:  () => api.get<{ count: number }>('/notifications/unread-count'),
  markRead:     (id: number) => api.post<{ status: string }>(`/notifications/${id}/read`, null),
  markAllRead:  () => api.post<{ status: string }>('/notifications/read-all', null),
  requestUpdate: (version: string, changelog: string) =>
    api.post<{ status: string; delayed: boolean; other_sessions: number }>(
      '/settings/request-update', { version, changelog },
    ),
}

File: backend/schemas/notifications.py
"""Схеми сповіщень у програмі."""

from typing import Optional
from pydantic import BaseModel


class NotificationOut(BaseModel):
    id:         int
    type:       str
    title:      str
    body:       Optional[str] = None
    meta:       Optional[str] = None
    created_at: str
    read_at:    Optional[str] = None

    model_config = {"from_attributes": True}


class UnreadCount(BaseModel):
    count: int


class RequestUpdateIn(BaseModel):
    version:   str
    changelog: str = ""
